modules/ROOT/pages/java-support.adoc

```diff
@@ -0,0 +1,174 @@
+
+[IMPORTANT]
+To ensure your API proxies or Mule apps are protected when upgrading, upgrade your API policies *before* upgrading your API proxies or Mule apps.
+
+[anypoint-connectors-and-modules]
+== Anypoint Connectors and Modules
+
+Anypoint Connectors and modules are certified by MuleSoft to run on Java 17. For information about Java 17 support for each connector and module, see the xref:release-notes::connector/anypoint-connector-release-notes.adoc[Release Notes] for that connector or module. Java compatibility information is also provided on the Exchange tiles for connectors and modules. 
+
+The most commonly used Anypoint Connectors and modules are Java 17-compatible now, and others are being released regularly. See https://help.salesforce.com/s/articleView?id=000782248&type=1[Java 17 Compatible Anypoint Connectors^] for a list of Java 17-compatible connectors.
+
+To upgrade Anypoint Connectors:
+
+. Update your Mule app to run on Mule Runtime 4.6 or later. 
+. Change the target JVM at the project property level. This includes both upgrades and downgrades of the JVM version.
+. In Anypoint Studio, update the connectors in your app to the latest Java 17-compatible version. 
+. Deploy your Mule application:
+  * xref:runtime-fabric::/deploy-to-runtime-fabric.adoc[Deploying Apps to Runtime Fabric]
+  * xref:runtime-manager::cloudhub/deploy-mule-application-task.adoc[Deploying Apps to CloudHub]
+  * xref:hosting::/ch2-deploy.adoc[Deploying Apps to CloudHub 2.0]
+
+[custom-connectors]
+=== Custom Connectors
+
+Custom connectors are any connectors that are not developed and maintained by MuleSoft, including connectors that are built by MuleSoft partners using Mule SDK, or XML SDK.
+
+If you are using a custom connector in your app, you must update your connector to run on Java 17 and MuleSoft Runtime 4.6 and later. After you update your connector using Mule Java SDK or XML SDK, you must test your app and deploy it. 
+
+image:java-upgrade-connectors.png[Java upgrade steps for connectors]
+
+[calloutlist]
+. Update your connector using:
+  * xref:mule-sdk/getting-started.adoc[Mule Java SDK]
+  * xref:mule-sdk/xml-sdk.adoc[Mule XML SDK]
+. Test your connector using xref:munit/index.adoc[MUnit].
+. Deploy your Mule application:
+  * xref:runtime-fabric::/deploy-to-runtime-fabric.adoc[Deploying Apps to Runtime Fabric]
+  * xref:runtime-manager::cloudhub/deploy-mule-application-task.adoc[Deploying Apps to CloudHub]
+  * xref:hosting::/ch2-deploy.adoc[Deploying Apps to CloudHub 2.0]
+
+[rest-connect-generated-connectors]
+=== REST Connect Generated Connectors
+
+For connectors that are generated from an API specification using the REST Connect connector generator tool, all you have to do to make the connector Java 17-compatible is republish the API specification to Exchange. 
```

Comment:
Daniel Vlad was saying we should mention how users can save time by activating TLS when rebuilding for J17 (similar to what is mentioned in the external J17 communication doc). Might need to add callout here for TLS